### PR TITLE
CORE-11905: Allow passing custom certificate as CRT file to REST Worker

### PR DESCRIPTION
**Pull request by vkolomeyko**

### CODE REVIEW

File: libs/rest/ssl-cert-read-impl/src/main/kotlin/net/corda/rest/ssl/impl/SslCertReadServiceImpl.kt [hunk 1]
```diff
@@ -55,15 +75,21 @@ class SslCertReadServiceImpl(private val createDirectory: () -> Path) : SslCertR
...
         if (localKeyStoreInfo != null) return localKeyStoreInfo
 
-        localKeyStoreInfo = if (config.hasPath(BootConfig.BOOT_REST_TLS_KEYSTORE_FILE_PATH)) {
-            val bootKeyStorePath = config.getString(BootConfig.BOOT_REST_TLS_KEYSTORE_FILE_PATH)
-            val keyStorePassword = config.getString(BootConfig.BOOT_REST_TLS_KEYSTORE_PASSWORD)
+        localKeyStoreInfo = if (config.hasPath(BOOT_REST_TLS_KEYSTORE_FILE_PATH)) {
+            val bootKeyStorePath = config.getString(BOOT_REST_TLS_KEYSTORE_FILE_PATH)
+            val keyStorePassword = config.getString(BOOT_REST_TLS_KEYSTORE_PASSWORD)
             KeyStoreInfo(Path.of(bootKeyStorePath), keyStorePassword)
+        } else if(config.hasPath(BOOT_REST_TLS_CRT_PATH)) {
```
Comment (vkolomeyko) [inline libs/rest/ssl-cert-read-impl/src/main/kotlin/net/corda/rest/ssl/impl/SslCertReadServiceImpl.kt:82]: The "meat" of the change.

File: libs/rest/ssl-cert-read-impl/src/main/kotlin/net/corda/rest/ssl/impl/SslCertReadServiceImpl.kt [hunk 2]
```diff
@@ -4,21 +4,41 @@ import net.corda.libs.configuration.SmartConfig
 import net.corda.rest.ssl.KeyStoreInfo
 import net.corda.rest.ssl.SslCertReadService
 import net.corda.schema.configuration.BootConfig
+import net.corda.schema.configuration.BootConfig.BOOT_REST_TLS_CA_CRT_PATH
+import net.corda.schema.configuration.BootConfig.BOOT_REST_TLS_CRT_PATH
+import net.corda.schema.configuration.BootConfig.BOOT_REST_TLS_KEYSTORE_FILE_PATH
+import net.corda.schema.configuration.BootConfig.BOOT_REST_TLS_KEYSTORE_PASSWORD
+import net.corda.schema.configuration.BootConfig.BOOT_REST_TLS_KEY_PATH
 import net.corda.v5.base.exceptions.CordaRuntimeException
+import org.bouncycastle.asn1.pkcs.PrivateKeyInfo
```
Comment (vkolomeyko) [inline libs/rest/ssl-cert-read-impl/src/main/kotlin/net/corda/rest/ssl/impl/SslCertReadServiceImpl.kt:13]: I have decided to use BouncyCastle library, as it saves some manual steps, like detecting an algorithm that was used for PrivateKey in PEM format.

**Review by davidcurrie — APPROVED** [review]
LGTM - I take it that Bouncy Castle is not a new dependency to C5 then?

**Review by filipesoliveira — APPROVED** [review]
LGTM

**Comment by vkolomeyko** [comment]
> LGTM - I take it that Bouncy Castle is not a new dependency to C5 then?

No, it is not. It is used by P2P and Crypto sub-systems already.

**Pull request merged**